hive-metastore/src/main/java/org/apache/iceberg/hive/HiveCatalog.java

```diff
@@ -539,12 +577,22 @@ Database convertToDatabase(Namespace namespace, Map<String, String> meta) {
             database.setDescription(value);
           } else if (key.equals("location")) {
             database.setLocationUri(value);
+          } else if (key.equals(HMS_DB_OWNER)) {
+            database.setOwnerName(value);
+          } else if (key.equals(HMS_DB_OWNER_TYPE) && value != null) {
+            database.setOwnerType(PrincipalType.valueOf(value));
           } else {
             if (value != null) {
               parameter.put(key, value);
             }
           }
         });
+
+    if (database.getOwnerName() == null) {
+      database.setOwnerName(System.getProperty("user.name"));
```

Comment:
You might want to consider falling back to `UserGroupInformation.getCurrentUser()` as opposed to just the java user.  I believe this is used in many placed by Spark and Hive to identify the current user and may be different from the process owner.

Reply:
Hi, its probably my fault here,  @haizhou-zhao had this initially but it seemed there was some unresolved concern at the time :  https://github.com/apache/iceberg/pull/6045#discussion_r1004221149   , so I actually asked @haizhou-zhao to follow what the table implementation uses for consistency , and we could change both to UserGroupInformation.getCurrentUser() later.  

But looks like its cleared up, @gaborkaszab ?  If so we can go ahead and use UserGroupInformation.getCurrentUser() here and then change table to be consistent as well?

Reply:
I think it actually makes sense to use just `user.name` in other places in Iceberg since we want to avoid hard dependencies on Hadoop/Hive where possible.  In this case though, Hive requires a Hadoop dependency and since this catalog is likely used in that context, `UserGroupInformation` seems like it would produce more consistent behavior with the Spark/Hive engine itself.

Other catalog implementation probably should not use UGI since they may be running on environments where hadoop is not available (e.g. some flink environments don't have hadoop).